pkg/generate/code/check.go

```diff
@@ -100,16 +104,18 @@ func checkRequiredFieldsMissingFromShape(
 	shape *awssdkmodel.Shape,
 ) string {
 	indent := strings.Repeat("\t", indentLevel)
-	if shape == nil || len(shape.Required) == 0 {
+	if shape == nil {
 		return fmt.Sprintf("%sreturn false", indent)
 	}
+	missing := []string{}
 
 	// Loop over the required member fields in the shape and identify whether
 	// the field exists in either the Status or the Spec of the resource and
 	// generate an if condition checking for all required fields having non-nil
 	// corresponding resource Spec/Status values
-	missing := []string{}
+	notRequiredFields := shape.MemberRefs
 	for _, memberName := range shape.Required {
+		delete(notRequiredFields, memberName)
```

Comment:
`delete` scares me - I think it's modifying `shape.MemberRefs`. Could we copy the map beforehand? https://stackoverflow.com/a/23058707/1278784

Reply:
Since go 1.11 it is totally safe to loop and delete a maps elements: https://golang.org/doc/go1.11#performance-compiler

Reply:
> Since go 1.11 it is totally safe to loop and delete a maps elements: https://golang.org/doc/go1.11#performance-compiler

I think @RedbackThomson was just concerned with mutating the aws-sdk-go Shape struct instead of a copy of the Shape's MemberRefs map.